colly.go

```diff
@@ -562,40 +561,28 @@ func (c *Collector) scrape(u, method string, depth int, requestData io.Reader, c
 	if err != nil {
 		return err
 	}
-	if err := c.requestCheck(u, parsedURL, method, requestData, depth, checkRevisit); err != nil {
-		return err
-	}
-
 	if hdr == nil {
 		hdr = http.Header{}
 	}
 	if _, ok := hdr["User-Agent"]; !ok {
 		hdr.Set("User-Agent", c.UserAgent)
 	}
-	rc, ok := requestData.(io.ReadCloser)
-	if !ok && requestData != nil {
-		rc = ioutil.NopCloser(requestData)
+	req, err := http.NewRequest(method, parsedURL.String(), requestData)
+	if err != nil {
+		return err
 	}
+	req.Header = hdr
 	// The Go HTTP API ignores "Host" in the headers, preferring the client
 	// to use the Host field on Request.
-	host := parsedURL.Host
 	if hostHeader := hdr.Get("Host"); hostHeader != "" {
-		host = hostHeader
-	}
-	req := &http.Request{
-		Method:     method,
-		URL:        parsedURL,
-		Proto:      "HTTP/1.1",
-		ProtoMajor: 1,
-		ProtoMinor: 1,
-		Header:     hdr,
-		Body:       rc,
-		Host:       host,
+		req.Host = hostHeader
 	}
 	// note: once 1.13 is minimum supported Go version,
 	// replace this with http.NewRequestWithContext
 	req = req.WithContext(c.Context)
-	setRequestBody(req, requestData)
+	if err := c.requestCheck(u, parsedURL, method, req.GetBody, depth, checkRevisit); err != nil {
```

Comment:
Probably I'm missing something, but I can't find where the `req.GetBody` function is implemented. Pls help. =]

Reply:
`GetBody` is set by `NewRequest`: https://cs.opensource.google/go/go/+/refs/tags/go1.17.8:src/net/http/request.go;l=889-930;drc=refs%2Ftags%2Fgo1.17.8

This code fragment used to be copy-paste in Colly (`setRequestBody`) and was removed in my previous commit.